app/assets/javascripts/discourse/app/lib/uppy-checksum-plugin.js

```diff
@@ -0,0 +1,93 @@
+import { Plugin } from "@uppy/core";
+import { Promise } from "rsvp";
+
+export default class UppyChecksum extends Plugin {
+  constructor(uppy, opts) {
+    super(uppy, opts);
+    opts = opts || {};
+    this.id = opts.id || "uppy-checksum";
+    this.capabilities = opts.capabilities;
+
+    // this is meaningless, can be anything
+    this.type = "preprocessor";
```

Comment:
Can we avoid setting it though so that we don't have to carry a random string and an additional comment.

Reply:
Nope, uppy errors and tells you you need a type. Removed the comment and just kept it as preprocessor